### DISCUSSION TITLE
Refactoring

### DISCUSSION THREAD

**Discussion by Gregory1701** [codefori/vscode-ibmi#1340] [IBM i]
Is there a way to refactor like in RDi?  For example, here is a snippet:

If sqlstate <> '00000';
  rows = 0;
Else;
  Exec SQL GET DIAGNOSTICS :rows = ROW_COUNT;
EndIf;

If I highlight '00000' and right click, I have a context menu option of 'Refactor...'.  This seems perfect as I would like to refactor this inline string into a constant at the top of my program.  When I choose the 'Refactor...' I receive the inline message stating that 'No refactorings available'.  The ideal resulting would be:

Dcl-c sql_success '00000';

If sqlstate <> sql_success;
  rows = 0;
Else;
  Exec SQL GET DIAGNOSTICS :rows = ROW_COUNT;
EndIf;

Is this possible?

**Comment by JameT55** [comment]
I see that "Refactor" doesn't work.  
I tried double-clicking on the word that I wanted to change and trying the "Change All Occurences" Ctrl+F2 option that appeared.
Every occurrence was highlighted.  As I changed the one that my cursor was on, all occurrences changes as well.
